### DISCUSSION TITLE
Can't connect to mssql server using info from SSMS

### DISCUSSION THREAD

**Discussion by refishokga** [sqlpage/SQLPage#244] [General]
Hi

I have a ms sql server at somehost.com,50135 (that's a comma before the port number) that works in ssms but I can't connect with sqlpage. 

I've tried both a comma and a colon in the config file, connect.rs says invalid port both ways. Default ssms settings are used, no other options are used as in encryption or certificates etc.

Any ideas?

![2024-02-27 09_26_22-Connect to Server](https://github.com/lovasoa/SQLpage/assets/161501237/3690a1a1-16db-4a88-8ed3-e7ce1df612cb)

**Comment by lovasoa** [comment]
Hi !
Is this an sqlexpress instance ? Did you try the instructions from https://github.com/lovasoa/SQLpage/issues/86 ?

Can you paste the EXACT error message you are getting ?

**Comment by refishokga** [comment]
This is mssql enterprise hosted on azure. This is not a named instance.

Exact error is
thread 'main' panicked at D:\a\SQLpage\SQLpage\src\webserver\database\connect.rs:16:34:
Invalid database URL: Configuration(InvalidPort)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by lovasoa** [comment]
You get this message when you specify a port that is not a number, or is not between 0 and 65535.
If your username or password contains special characters (a `/` in particular), you should make sure to percent-encode them so that the string is interpreted correctly, and sqlpage doesn't try to interpret part of your password as the port, for instance. 

### Example 

If your username is `yellow:submarine`, your password is `the@beatles`, your host is `example.com`, your port is 50135, and your database is `hello/world`, the connection string will be:

```url
mssql://yellow%3Asubmarine:the%40beatles@example.com:50135/hello%2Fworld
```

**Reply by refishokga** [reply]
Ah ha! I must have missed the part about url encoding my fields. Done and now it's up and running. Thanks for the assistance.

**Reply by lovasoa** [reply]
Great! 

I hope you'll share the cool app you built with SQLPage here soon!

I'll update the documentation about connection strings.

**Comment by castlekaybee** [comment]
Sorry to bump this but it's related ;)

I can connect to my MSSQL Server through e.g. DBGate with the IP adress on standard port 1433 (screenshot).

sqlpage however gives me an error message "invalid port number". I have read and tried [#86](https://github.com/lovasoa/SQLpage/issues/86) but in my case I don't have a named instance and the port seems to reply fine. 

What could cause this?
![image](https://github.com/lovasoa/SQLpage/assets/145761957/16079c23-bad0-41cc-8341-a9b3c3d923f2)
![image](https://github.com/lovasoa/SQLpage/assets/145761957/9f576326-8d95-4ef0-a03d-b5462fe07c16)

**Reply by castlekaybee** [reply]
this is sqlpage windows executable

**Reply by lovasoa** [reply]
Hello and welcome to SQLPage!

It's hard to say where the problem is in the database configuration string without seeing it, but if I had to guess, I would say that it's the password that contains a special character that you forgot to percent-encode.  

See the documentation here: https://github.com/lovasoa/SQLpage/blob/main/configuration.md#database-connection-strings

For instance, if your password is `a:b` it should be encoded as `a%3Ab`




Out of curiosity: what is it that you are building?

**Reply by castlekaybee** [reply]
Oh sure. Thanks, it works. The error returned is of course misleading.
It's a frontend for my companys portfolio of machines.

It's an awesome project.
